app/server/appsmith-interfaces/src/main/java/com/appsmith/external/helpers/restApiUtils/helpers/RestAPIActivateUtils.java

```diff
@@ -213,6 +226,20 @@ protected Mono<ClientResponse> httpCall(
                 });
     }
 
+    public URI createRedirectUrl(String redirectUrl, URI originalUrl) throws URISyntaxException {
+        URI redirectUri = new URI(redirectUrl);
+        String scheme =
+                StringUtils.hasText(redirectUri.getScheme()) ? redirectUri.getScheme() : originalUrl.getScheme();
+        String authority = StringUtils.hasText(redirectUri.getAuthority())
+                ? redirectUri.getAuthority()
+                : originalUrl.getAuthority();
+        String path = redirectUri.getPath();
+        String query = redirectUri.getQuery();
+        String fragment = redirectUri.getFragment();
+
+        return new URI(scheme, authority, path, query, fragment);
+    }
```

Comment:
Can we use the [`URI.resolve`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/net/URI.html#resolve(java.lang.String)) method to do the resolution here instead of manually doing it please? It'll handle few other cases correctly, like `../redirect-path` etc.

```suggestion
    public URI createRedirectUrl(String redirectUrl, URI originalUrl) {
        return originalUrl.resolve(redirectUrl);
    }
```

Reply:
got it, making the change